### PR TITLE
added https option for repository

### LINKED ISSUE #70: Enable https for Bareos repos
In https://github.com/voxpupuli/puppet-bareos/blob/master/manifests/repository.pp#L28 plain http is used for downloading the bareos packages.
This should be changed to https or it should be at least the default.
Both https://download.bareos.org and https://download.bareos.com are available via https

### PR DESCRIPTION
**Pull request by benibr**
#### Pull Request (PR) description
added a parameter which enables https by default. can be set to false to use http without TLS

#### This Pull Request (PR) fixes the following issues
Fixes #70

### CODE REVIEW

File: manifests/repository.pp [hunk 1]
```diff
@@ -17,15 +17,22 @@
...
 class bareos::repository (
   String              $release             = 'latest',
   Optional[String[1]] $gpg_key_fingerprint = undef,
   Boolean             $subscription        = false,
   Optional[String]    $username            = undef,
   Optional[String]    $password            = undef,
+  Boolean             $https               = true,
 ) {
-  $scheme = 'http://'
+  if $https {
```
Comment (ghoneycutt) [inline manifests/repository.pp:31]: While your code works, this is more clear to the reader.

```puppet
if $https == true {
```

File: manifests/repository.pp [hunk 2]
```diff
@@ -17,15 +17,22 @@
...
+# @param https
+#   Wether or not https should be used in repo URL
 #
 class bareos::repository (
   String              $release             = 'latest',
   Optional[String[1]] $gpg_key_fingerprint = undef,
   Boolean             $subscription        = false,
   Optional[String]    $username            = undef,
   Optional[String]    $password            = undef,
+  Boolean             $https               = true,
```
Comment (ghoneycutt) [inline manifests/repository.pp:29]: This needs a spec test that shows the default case as well as what happens when the parameter is set to `false`.

**Review by ghoneycutt — CHANGES_REQUESTED** [review]

File: manifests/repository.pp (hunk 2 above)
Comment (ghoneycutt) [reply manifests/repository.pp:29]: The default should, very unfortunately, stay `http` as this is a breaking change to modify the behavior by changing the protocol. If you are doing a lot of work with this module, you might get feedback about any other wanted breaking changes and coordinate to get those in and a major version bump.

File: manifests/repository.pp (hunk 1 above)
Comment (kenyon) [reply manifests/repository.pp:31]: @ghoneycutt that's entirely redundant, nonstandard (for any language, not just puppet), and not supported by the Style Guide or other documentation. IMO it's an antipattern, redundant code should be avoided. More code means more potential for problems.

Comment (ghoneycutt) [reply manifests/repository.pp:31]: Having spent a lot of time teaching people with no coding skills how to model their infrastructure with Puppet, this helps them. Anyhow, it was just an observation and not meant as something to be done to pass the review.

File: manifests/repository.pp [hunk 3]
```diff
@@ -17,15 +17,22 @@
 #   The major bareos release version which should be used
 # @param password
 #   The major bareos release version which should be used
+# @param https
+#   Wether or not https should be used in repo URL
```
Comment (kenyon) [inline manifests/repository.pp:21]:
```suggestion
#   Whether https should be used in repo URL
```

**Review by kenyon — CHANGES_REQUESTED** [review]

**Comment by jhoblitt** [comment]
We are about to make a major release and this seems like a good time to make https the default.

**Comment by jhoblitt** [comment]
@benibr Thank you for sticking with this. Merged as #123.

**Pull request closed without merging**